File: vlsisapd/examples/dtr/cplusplus/parseDtr.cpp
#include <iostream>
#include <string>
using namespace std;

#include "vlsisapd/dtr/Techno.h"
#include "vlsisapd/dtr/Name.h"

int main(int argc, char * argv[]) {
    DTR::Techno* techno = DTR::Techno::readFromFile("./example.dtr.xml");

    cerr << "+-----------------------------+" << endl
         << "| technology:      " << techno->getName().getString() << "    |"   << endl
         << "| units:           " << techno->getUnit().getString() << "      |" << endl
         << "+-----------------------------+" << endl << endl;

    cerr << "transistorMinL                = " << techno->getValue(DTR::Name("transistorMinL")) << endl
         << "transistorMinW                = " << techno->getValueAsString(DTR::Name("transistorMinW")) << endl
         << "minWidth of metal1            = " << techno->getValue(DTR::Name("minWidth"), DTR::Name("metal1")) << endl
         << "minSpacing of metal1          = " << techno->getValue(DTR::Name("minWidth"), DTR::Name("metal1")) << endl
         << "minSpacing of active vs poly  = " << techno->getValue(DTR::Name("minSpacing"), DTR::Name("active"), DTR::Name("poly")) << endl
         << "minExtension active over poly = " << techno->getValue(DTR::Name("minExtension"), DTR::Name("poly"), DTR::Name("active")) << endl
         << "minArea of metal1             = " << techno->getValue(DTR::Name("minArea"), DTR::Name("metal1")) << endl;

    return 0;
}

File: vlsisapd/examples/dtr/python/parseDtr.py
from DTR import *
from decimal import Decimal

techno = Techno.readFromFile("./example.dtr.xml")

print "+-----------------------------+"
print "| technology:      "+techno.getName().getString()+"    |"
print "| units:           "+techno.getUnit().getString()+"      |"
print "| version:         "+techno.getVersion().getString()+"      |"
print "+-----------------------------+\n\n"

print "transistorMinL                = %s"%techno.getValue(Name("transistorMinL"))
print "transistorMinW                = %s"%Decimal(techno.getValueAsString(Name("transistorMinW")))
print "minWidth of metal1            = %s"%techno.getValue(Name("minWidth"), Name("metal1"))
print "minSpacing of metal1          = %s"%techno.getValue(Name("minWidth"), Name("metal1"))
print "minSpacing of active vs poly  = %s"%techno.getValue(Name("minSpacing"), Name("active"), Name("poly"))
print "minExtension active over poly = %s"%techno.getValue(Name("minExtension"), Name("poly"), Name("active"))
print "minArea of metal1             = %s"%techno.getValue(Name("minArea"), Name("metal1"))

# an example of why it is important to use Decimal in python:
print techno.getValue(Name("minArea"), Name("metal1"))*3-0.3 # returns 5.55111512313e-17
print Decimal(techno.getValueAsString(Name("minArea"), Name("metal1")))*3-Decimal('0.3') # returns 0.000
